packages/core/strapi/lib/utils/cron.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const shiftHours = (date, step) => {
+  const frequency = 24 / step;
+  const list = Array.from({ length: frequency }, (_, index) => index * step);
+  const hour = date.getHours();
+  return list.map((value) => (value + hour) % 24).sort((a, b) => a - b);
+};
+
+// TODO: This should be transformed into a cron expression shifter that could be reused in other places
+// For now it's tailored to the license check cron, scheduled every 12h
+const getRecurringCronExpression = (date) => `${date.getMinutes()} ${shiftHours(date, 12)} * * *`;
```

Comment:
It's not what we need as the initial goal was to emulate an interval - `node-schedule` is not well suited for that.
The `startTime` and `endTime` options are useful if you want to define a range of time a cron will execute. We want it to run indefinitely here.

We shift the minutes and hours a little to ensure that not all projects in the same timezone check their license simultaneously. This could be prevented differently (randomize instead of shift for instance).

Reply:
Ok I missunderstood the goal here. I though you wanted to launch the cron x hours after the start of the process and then every x hours. you can do that using only the startTime